src/Markdig/Parsers/HeadingBlockParser.cs

```diff
@@ -95,26 +95,23 @@ public override BlockState TryOpen(BlockProcessor processor)
                         }
                         endState = 1;
                     }
-                    if (endState == 1)
```

Comment:
Removing this change completely the meaning of the loop. This is not a valid change.